packages/common/src/actions/transactionQueue.ts

```diff
@@ -0,0 +1,14 @@
+import type { Transport, Chain, Account, WalletActions, WalletClient } from "viem";
+import { writeContract as mud_writeContract } from "../writeContract";
+import { sendTransaction as mud_sendTransaction } from "../sendTransaction";
+
+export const transactionQueue = <TChain extends Chain, TAccount extends Account>(): ((
+  client: WalletClient<Transport, TChain, TAccount>
+) => Pick<WalletActions<TChain, TAccount>, "writeContract" | "sendTransaction">) => {
+  return (client) => ({
+    // Applies to: `client.writeContract`, `getContract(client, ...).write`
+    writeContract: (args) => mud_writeContract(client, args),
+    // Applies to: `client.sendTransaction`
+    sendTransaction: (args) => mud_sendTransaction(client, args),
```

Comment:
If I said we wanted to eventually get rid of `writeContract` and `sendTransaction` "wrappers" in favor of these actions, would you think it better to move the logic in here now?

In that case, I _think_ we'd only need to implement `sendTransaction`? Since `writeContract` internally wraps `sendTransaction`.

Reply:
> would you think it better to move the logic in here

We don't need the logic exactly here, but by not exposing `mud_sendTransaction`, I think I can remove a few lines of code:

https://github.com/latticexyz/mud/blob/db314a7490b7797322fd0568cbeec0066c231666/packages/common/src/sendTransaction.ts#L28-L33

> we'd only need to implement sendTransaction?

I wouldn't do so. Since we include simulation, we want to use `call` for sendTransaction and `simulateContract` for writeContract for the best error revert message.

Reply:
oh right! I forgot that's why they were done separately